README.md
# EHPlainAlert

This is simple extension for presenting multiple system-wide notifications from bottom of device screen.

<img src="http://josshad.github.io/EHPlainAlert/EHPlainAlert.gif">

## Requirements

- Requires iOS 7.1 or later
- Requires Automatic Reference Counting (ARC)

##Features

- Supports multiple messages on one screen
- Simple use actions
- Highly customizable
- Call from anywhere in app

## Installation

### CocoaPods
To install EHPlainAlert using CocoaPods, please integrate it in your existing Podfile, or create a new Podfile:

```ruby
platform :ios, '7.1'

target 'MyApp' do
  pod 'EHPlainAlert'
end
```
Then run `pod install`.

### Manual

Add EHPlainAlert folder to your project 


##Usage

	#import <EHPlainAlert/EHPlainAlert.h>

###Presenting notification

All messages can simply presented via static method call:

	[EHPlainAlert showAlertWithTitle:@"Success" message:@"Something works!" type:ViewAlertSuccess];

Messages can be displayed from any location in app, even not associated with UI. 

    [[NetHelper sharedInstance] postRequestWithURLString:URL data:data withSuccess:^(id responseObject) {
            [EHPlainAlert showAlertWithTitle:@"Success" message:@"Data successfully uploaded" type:ViewAlertSuccess];
        } failure:^(NSError *error)
        {
            [EHPlainAlert showError:error];
        }];


###Presenting error notification

For simplifying error handling you can use:

	- (void)someError:(NSError *)myError
	{
		[EHPlainAlert showError:error];
	}

###Hiding Messages

Notifications will hidden automatically after 4 seconds. 

To change default delay use static method:
    [EHPlainAlert updateHidingDelay:2.5f];

Also you can just tap on message to hide it.

###On Tap Actions

You can change the default behavior for tapping on the notification:

    EHPlainAlert * ehAlert = [[EHPlainAlert alloc] initWithTitle:@"Hmm..." message:@"Tap for information" type:ViewAlertInfo];
    ehAlert.action = ^{
        [[UIApplication sharedApplication] openURL:[NSURL URLWithString:@"https://github.com/josshad/EHPlainAlert"]];
    };
    [ehAlert show];

###Customization

####Fonts:

Change font of one alert

	EHPlainAlert * ehAlert = [[EHPlainAlert alloc] initWithTitle:@"Info" message:@"This is info message" type:ViewAlertInfo];
    ehAlert.titleFont = [UIFont fontWithName:@"TrebuchetMS" size:15];
    ehAlert.subTitleFont = [UIFont fontWithName:@"TrebuchetMS-Italic" size:12];
    [ehAlert show];

Change fonts for alert type

    [EHPlainAlert updateTitleFont:[UIFont fontWithName:@"TrebuchetMS" size:18]];
    [EHPlainAlert updateSubTitleFont:[UIFont fontWithName:@"TrebuchetMS" size:10]];

####Colors:

Change background color of one alert

	EHPlainAlert * ehAlert = [[EHPlainAlert alloc] initWithTitle:@"Hmm..." message:@"Blue color alert" type:ViewAlertInfo];
    ehAlert.messageColor = [UIColor blueColor];
    [ehAlert show];

Change color for alert type

    [EHPlainAlert updateAlertColor:[UIColor colorWithWhite:0 alpha:0.5] forType:ViewAlertPanic];

####Appearance

    [EHPlainAlert updateAlertPosition:ViewAlertPositionTop];

####Icons:

Change icon of one alert

    EHPlainAlert * ehAlert = [[EHPlainAlert alloc] initWithTitle:@"Hmm..." message:@"Blue color alert" type:ViewAlertInfo];
    ehAlert.iconImage = image;
    [ehAlert show];

Change icon for alert type

    [EHPlainAlert updateAlertIcon:image forType:ViewAlertInfo]; 

####Number of messages
    
    [EHPlainAlert updateNumberOfAlerts:4];

####Display time

    [EHPlainAlert updateHidingDelay:2.5f];

##Author
Danila Gusev

<a href="mailto:jos.shad@gmail.com">jos.shad@gmail.com</a>

## License

Usage is provided under the <a href="http://opensource.org/licenses/MIT" target="_blank">MIT</a> License. See <a href="https://github.com/josshad/EHPlainAlert/blob/master/LICENSE">LICENSE</a> for full details.


CHANGELOG.md
### Version 1.2.5
methods added:
	+ (void)updateShouldHideOnTap:(BOOL)hide;
	+ (void)updateShouldShowCloseIcon:(BOOL)show;
properties added:
	@property (nonatomic, assign) EHBoolean shouldHideOnTap;
	@property (nonatomic, assign) EHBoolean shouldShowCloseIcon;


### Version 1.2.5
methods added:
	-hide
	+hideAll:animated


### Version 1.2.1
Test updated

### Version 1.2.0
Added appearance customization

### Version 1.1.1
Fixing pods bundle bug

### Version 1.1.0
Icons customization

### Version 1.0.6
Static methods for changing default behavior.
	+ (void)updateNumberOfAlerts:(NSInteger)numberOfAlerts;
	+ (void)updateHidingDelay:(float)delay;
	+ (void)updateTitleFont:(UIFont *)titleFont;
	+ (void)updateSubTitleFont:(UIFont *)subtitleFont;
	+ (void)updateAlertColor:(UIColor *)color forType:(ViewAlertType)type;

### Version 1.0.1
* Added Changelog

## Version 1.0

* Customisation for colors, icons, fonts, action blocks, number of onscreen messages
* Simple presenting errors
* XCTest


EHPlainAlert/EHPlainAlert.h
//
//  EHPlainAlert.h
//  HMTest
//
//  Created by Danila Gusev on 09/10/15.
//  Copyright © 2015 josshad. All rights reserved.
//

#import <UIKit/UIKit.h>


typedef enum : NSUInteger {
    ViewAlertError,
    ViewAlertSuccess,
    ViewAlertInfo,
    ViewAlertPanic,
    ViewAlertUnknown
} ViewAlertType;

typedef enum : NSUInteger {
    ViewAlertPositionBottom = 0,
    ViewAlertPositionTop
} ViewAlertPosition;

typedef void (^ ActionBlock)();


@interface EHPlainAlert : UIViewController

/*!
 * @brief Block that is invoked on notification tap
 *
 */
@property (nonatomic, strong) ActionBlock action;

/*!
 * @brief Font for a title message. Default font: [UIFont fontWithName:@"HelveticaNeue-Light" size:15]
 *
 */
@property (nonatomic, strong) UIFont * titleFont;

/*!
 * @brief Font for a subtitle message. Default font: [UIFont fontWithName:@"HelveticaNeue-Light" size:12]
 *
 */
@property (nonatomic, strong) UIFont * subTitleFont;

/*!
 * @brief Color of notification.
 *
 */
@property (nonatomic, strong) UIColor * messageColor;

/*!
 * @brief Icon of the notification.
 *
 */
@property (nonatomic, strong) UIImage * iconImage;

/*!
 * @brief Title of the notification;
 *
 */
@property (nonatomic, copy) NSString * titleString;

/*!
 * @brief Subtitle of the notification;
 *
 */
@property (nonatomic, copy) NSString * subtitleString;

/*!
 * @brief show notification with title "Error" and subtitle error.localizedDescription
 *
 * @param error Error to show in alert
 */
+ (instancetype)showError:(NSError *)error;

/*!
 * @brief show notification with title error.domain and subtitle error.localizedDescription
 *
 * @param error Error to show in alert
 */
+ (instancetype)showDomainError:(NSError *)error;

/*!
 * @brief show default notification
 *
 * @param title Title of the notification
 *
 * @param message Subtitle of the notification
 *
 * @param type The type of the notification, e.g. ViewAlertSuccess
 */
+ (instancetype)showAlertWithTitle:(NSString *)title message:(NSString *)message type:(ViewAlertType)type;

/*!
 * @brief hide all alerts
 *
 * @param animated use animation
 */
+(void)hideAll:(BOOL)animated;

/*!
 * @brief initialize notification
 *
 * @param title Title of the notification
 *
 * @param message Subtitle of the notification
 *
 * @param type The type of the notification, e.g. ViewAlertSuccess
 */
- (id)initWithTitle:(NSString *)title message:(NSString *)message type:(ViewAlertType)type;


/*!
 * @brief show notification
 */
- (void)show;

/*!
 * @brief hide notification
 */
- (void)hide;
/*!
 * @brief change maximum number of alerts on screen
 *
 * @param numberOfAlerts Number of alertviews. Default is 3
 */
+ (void)updateNumberOfAlerts:(NSInteger)numberOfAlerts;

/*!
 * @brief change onscreen time of alertview
 *
 * @param delay Time after which an alert is automatically hidden. Default is 4
 */
+ (void)updateHidingDelay:(float)delay;

/*!
 * @brief change default title font
 *
 * @param titleFont Title font. Default is HelveticaNeue-Light 15
 */
+ (void)updateTitleFont:(UIFont *)titleFont;

/*!
 * @brief change default message font
 *
 * @param titleFont Message font. Default is HelveticaNeue-Light 12
 */
+ (void)updateSubTitleFont:(UIFont *)subtitleFont;

/*!
 * @brief change default alert positon
 *
 * @param viewPosition of type ViewAlertPosition. Default is ViewAlertPositionBottom
 */
+ (void)updateAlertPosition:(ViewAlertPosition)viewPosition;

/*!
 * @brief change default alert color
 *
 * @param color Background color
 *
 * @param type Message type
 */
+ (void)updateAlertColor:(UIColor *)color forType:(ViewAlertType)type;

/*!
 * @brief change default alert icon
 *
 * @param image Icon image
 *
 * @param type Message type
 */
+ (void)updateAlertIcon:(UIImage *)image forType:(ViewAlertType)type;

/*!
 * @brief change default alert icon
 *
 * @param image Icon image
 *
 * @param type Message type
 */

@end


EHPlainAlert/EHPlainAlert.m
//
//  EHPlainAlert.m
//  HMTest
//
//  Created by Danila Gusev on 09/10/15.
//  Copyright © 2015 josshad. All rights reserved.
//

#import "EHPlainAlert.h"
#import "UIColor+HexRGB.h"

#define EHDEFAULT_TITLE_FONT [UIFont fontWithName:@"HelveticaNeue-Light" size:15]
#define EHDEFAULT_SUBTITLE_FONT [UIFont fontWithName:@"HelveticaNeue-Light" size:12]
#define EHDEFAULT_MAX_ALERTS_NUMBER 3
#define EHDEFAULT_HIDING_DELAY 4
#define EHPOSITIONFORALERT(i) (_EHAlertPosition == ViewAlertPositionBottom)?screenSize.height - 70 * (i + 1) - 0.5 * (i):70 * (i) + 0.5 * (i)

static NSInteger _EHNumberOfVisibleAlerts = EHDEFAULT_MAX_ALERTS_NUMBER;
static ViewAlertPosition _EHAlertPosition = ViewAlertPositionBottom;
static float _EHHidingDelay = EHDEFAULT_HIDING_DELAY;
static UIFont * _EHTitleFont = nil;
static UIFont * _EHSubTitleFont = nil;

static NSMutableDictionary * _EHColorsDictionary = nil;
static NSMutableDictionary * _EHIconsDictionary = nil;

float EH_iOS_Version() {
    return [[[UIDevice currentDevice] systemVersion] floatValue];
}

@interface NSBundle (ios7Bundle)

@end

@implementation NSBundle (ios7Bundle)

+ (instancetype)ios7Bundle{
    NSBundle *mainBundle = [NSBundle mainBundle];
    NSURL *bundleUrl = [mainBundle URLForResource:@"EHPlainAlert" withExtension:@"EHPlainAlert"];
    NSBundle *bundle = [NSBundle bundleWithURL:bundleUrl];
    return bundle;
}

+ (UIImage*)imageNamed:(NSString*)name{
    UIImage *image;
    
    image = [UIImage imageNamed:[NSString stringWithFormat:@"EHPlainAlert.bundle/%@",name]];
    if (image) {
        return image;
    }
    
    image = [UIImage imageWithContentsOfFile:[[[NSBundle ios7Bundle] resourcePath] stringByAppendingPathComponent:name]];
    
    return image;
}
@end

@implementation EHPlainAlert
{
    CGSize screenSize;
    ViewAlertType _alertType;
    BOOL _iconSetted;
}

static NSMutableArray * currentAlertArray = nil;

+ (instancetype)showError:(NSError *)error
{
    return [self showAlertWithTitle:@"Error" message:error.localizedDescription type:ViewAlertError];
}


+ (instancetype)showDomainError:(NSError *)error
{
    return [self showAlertWithTitle:error.domain message:error.localizedDescription type:ViewAlertError];
}


+ (instancetype)showAlertWithTitle:(NSString *)title message:(NSString *)message type:(ViewAlertType)type
{
    EHPlainAlert * alert = [[EHPlainAlert alloc] initWithTitle:title message:message type:type];
    [alert show];
    return alert;
}

+(void)hideAll:(BOOL)animated
{
    for (EHPlainAlert * alert in currentAlertArray)
    {
        [alert hide:@(animated)];
    }
}

- (id)initWithTitle:(NSString *)title message:(NSString *)message type:(ViewAlertType)type;
{
    self = [super init];
    if (self)
    {
        self.titleString = title;
        self.subtitleString = message;
        if (!currentAlertArray)
        {
            currentAlertArray = [NSMutableArray new];
        }
        [EHPlainAlert  updateColorsDictionary];
        [EHPlainAlert  updateIconsDictionary];
        _alertType = type;
    }
    return self;
}

- (instancetype)init{
    self = [super init];
    if (self)
    {
        if (!currentAlertArray)
        {
            currentAlertArray = [NSMutableArray new];
        }
        [EHPlainAlert updateColorsDictionary];
        [EHPlainAlert  updateIconsDictionary];
    }
    return self;
}

+ (void)updateColorsDictionary
{
    if (!_EHColorsDictionary)
    {
        _EHColorsDictionary = [@{ @(ViewAlertError) : [UIColor colorWithHex:@"#FDB937"],
                                  @(ViewAlertSuccess) : [UIColor colorWithHex:@"#49BB7B"],
                                  @(ViewAlertInfo) :  [UIColor colorWithHex:@"#00B2F4"],
                                  @(ViewAlertPanic) :[UIColor colorWithHex:@"#f24841"]
                                  } mutableCopy];
    }
}

+ (void)updateIconsDictionary
{
    if (!_EHIconsDictionary)
    {
        _EHIconsDictionary = [@{ @(ViewAlertError) : [EHPlainAlert imageNamed:@"eh_alert_error_icon"],
                                  @(ViewAlertSuccess) : [EHPlainAlert imageNamed:@"eh_alert_complete_icon"],
                                  @(ViewAlertInfo) :  [EHPlainAlert imageNamed:@"eh_alert_info_icon"],
                                  @(ViewAlertPanic) :[EHPlainAlert imageNamed:@"eh_alert_error_icon"]
                                  } mutableCopy];
    }
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    if (!_EHTitleFont)
    {
        _EHTitleFont = EHDEFAULT_TITLE_FONT;
    }
    if (!_EHSubTitleFont)
    {
        _EHSubTitleFont = EHDEFAULT_SUBTITLE_FONT;
    }
    
    self.view.backgroundColor = [UIColor clearColor];
    
    screenSize = [UIScreen mainScreen].bounds.size;
    self.view.frame = CGRectMake(0, (_EHAlertPosition)?-70:screenSize.height, screenSize.width , 70);
    self.view.layer.masksToBounds = NO;
    
    [self constructAlert];
    
    UITapGestureRecognizer * tapGesture = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(onTap)];
    [self.view addGestureRecognizer:tapGesture];
}

- (void)constructAlert
{
    UIView * infoView = [UIView new];
    infoView.frame = CGRectMake(0, 0, self.view.bounds.size.width , 70);
    
    
    [self.view addSubview:infoView];
    
    UILabel * titleLabel = [[UILabel alloc] initWithFrame:CGRectMake(55, 0, infoView.frame.size.width - 70, infoView.frame.size.height)];
    
    titleLabel.backgroundColor = [UIColor clearColor];
    titleLabel.textAlignment = NSTextAlignmentLeft;
    titleLabel.textColor = [UIColor whiteColor];
    titleLabel.numberOfLines = 0;
    [infoView addSubview:titleLabel];
    
    NSMutableAttributedString * titleString = [[NSMutableAttributedString alloc] initWithString:_titleString ? _titleString : @""
                                                                                     attributes:@{NSFontAttributeName : _titleFont ? _titleFont : _EHTitleFont}];
    
    NSAttributedString * messageString = [[NSAttributedString alloc] initWithString:[NSString stringWithFormat:@"\n%@",_subtitleString ? _subtitleString : @""]
                                                                         attributes:@{NSFontAttributeName : _subTitleFont ? _subTitleFont : _EHSubTitleFont}];
    
    [titleString appendAttributedString:messageString];
    
    titleLabel.attributedText = titleString;
    
    UIImageView * imageView = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, 50, 70)];
    UIColor * bgColor = [_EHColorsDictionary objectForKey:@(_alertType)];
    if (!_iconSetted)
        _iconImage = [_EHIconsDictionary objectForKey:@(_alertType)];
 
    if (!bgColor)
    {
        bgColor = [UIColor colorWithHex:@"#FDB937"];
    }
 
    
    infoView.backgroundColor = _messageColor ? _messageColor : bgColor;
    imageView.image = _iconImage;   
    imageView.contentMode = UIViewContentModeCenter;
    [infoView addSubview:imageView];
    
    UIImageView * closeView = [[UIImageView alloc] initWithImage:[EHPlainAlert imageNamed:@"eh_alert_close_icon"]];
    closeView.frame = CGRectMake(infoView.bounds.size.width - 15, 8, 7, 7);
    closeView.contentMode = UIViewContentModeCenter;
    
    closeView.userInteractionEnabled = YES;
    [infoView addSubview:closeView];
    
    UITapGestureRecognizer * tapGesture = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(onCloseTap:)];
    [closeView addGestureRecognizer:tapGesture];
}

+ (UIImage *)imageNamed:(NSString *)name
{
    if (EH_iOS_Version() < 8)
    {
        return [NSBundle imageNamed:name];
    }
    else
    {
        NSBundle * pbundle = [NSBundle bundleForClass:[self class]];
        NSString *bundleURL = [pbundle pathForResource:@"EHPlainAlert" ofType:@"bundle"];
        NSBundle *imagesBundle = [NSBundle bundleWithPath:bundleURL];
        UIImage * image = [UIImage imageNamed:name inBundle:imagesBundle compatibleWithTraitCollection:nil];
        return image;
    }
}

- (void)show
{
    [self performSelectorOnMainThread:@selector(showInMain) withObject:nil waitUntilDone:NO];
}

- (void)showInMain
{
    @synchronized (currentAlertArray) {
    
        if ([currentAlertArray count] == _EHNumberOfVisibleAlerts)
        {
            [[currentAlertArray firstObject] hide:@(YES)];
        }
        
        NSInteger numberOfAlerts = [currentAlertArray count];
        if (numberOfAlerts == 0)
            [([UIApplication sharedApplication].delegate).window addSubview:self.view];
        else
            [([UIApplication sharedApplication].delegate).window insertSubview:self.view belowSubview:[((EHPlainAlert *)[currentAlertArray lastObject]) view]];
        [UIView animateWithDuration:0.3 animations:^{
            self.view.frame = CGRectMake(0, EHPOSITIONFORALERT(numberOfAlerts), screenSize.width, 70);
        }];
        
        [currentAlertArray addObject:self];
        
        [self performSelector:@selector(hide:) withObject:@(YES) afterDelay:_EHHidingDelay];
        
    }
}

- (void)hide:(NSNumber *)nAnimated
{
    [self performSelectorOnMainThread:@selector(hideInMain:) withObject:nAnimated waitUntilDone:NO];
}

- (void)hideInMain:(NSNumber *)nAnimated
{
    @synchronized (currentAlertArray) {
        [currentAlertArray removeObject:self];
        BOOL animated = [nAnimated boolValue];
        if (animated)
        {
            [UIView animateWithDuration:0.5 animations:^{
                self.view.alpha = 0.7;
                self.view.frame = CGRectMake(0, (_EHAlertPosition)?-70:screenSize.height, screenSize.width , 70);
            } completion:^(BOOL finished) {
                [self.view removeFromSuperview];
            }];
            
            for (int i = 0; i < [currentAlertArray count]; i++)
            {
                EHPlainAlert * alert = [currentAlertArray objectAtIndex:i];
                [UIView animateWithDuration:0.5 animations:^{
                    alert.view.frame = CGRectMake(0, EHPOSITIONFORALERT(i), screenSize.width, 70);
                }];
            }
        }
        else
        {
            [self.view removeFromSuperview];
            for (int i = 0; i < [currentAlertArray count]; i++)
            {
                EHPlainAlert * alert = [currentAlertArray objectAtIndex:i];
                alert.view.frame = CGRectMake(0, EHPOSITIONFORALERT(i), screenSize.width, 70);
            }
        }
    }
}

- (void)hide
{
    [self hide:@(YES)];
}

- (void)onTap
{
    [self hide];
    
    if (_action != nil)
    {
        _action();
    }
}

- (void)onCloseTap:(UIGestureRecognizer *)gesture
{
    [self hide];
}

#pragma mark - Setters
- (void)setIconImage:(UIImage *)iconImage
{
    _iconImage = iconImage;
    _iconSetted = YES;
}

#pragma mark - Default behaviour

+ (void)updateNumberOfAlerts:(NSInteger)numberOfAlerts
{
    if (numberOfAlerts > 0)
    {
        _EHNumberOfVisibleAlerts = numberOfAlerts;
    }
}

+ (void)updateHidingDelay:(float)delay
{
    if (delay >= 0)
    {
        _EHHidingDelay = delay;
    }
}

+ (void)updateTitleFont:(UIFont *)titleFont
{
    _EHTitleFont = titleFont;
}

+ (void)updateSubTitleFont:(UIFont *)stitleFont
{
    _EHSubTitleFont = stitleFont;
}

+ (void)updateAlertPosition:(ViewAlertPosition)viewPosition
{
    _EHAlertPosition = viewPosition;
}

+ (void)updateAlertColor:(UIColor *)color forType:(ViewAlertType)type
{
    [EHPlainAlert updateColorsDictionary];
    if (color)
    {
        [_EHColorsDictionary setObject:color forKey:@(type)];
    }
    else
    {
        [_EHColorsDictionary removeObjectForKey:@(type)];
    }
}


+ (void)updateAlertIcon:(UIImage *)image forType:(ViewAlertType)type
{
    [EHPlainAlert updateIconsDictionary];
    if (image)
    {
        [_EHIconsDictionary setObject:image forKey:@(type)];
    }
    else
    {
        [_EHIconsDictionary removeObjectForKey:@(type)];
    }
}
@end


EHPlainAlertExampleTests/EHPlainAlertExampleTests.m
//
//  EHPlainAlertExampleTests.m
//  EHPlainAlertExampleTests
//
//  Created by Danila Gusev on 28/03/2016.
//  Copyright © 2016 josshad. All rights reserved.
//

#import <XCTest/XCTest.h>
#import "EHPlainAlert.h"
#import "UIColor+EHColorAdditions.h"

@interface EHPlainAlertExampleTests : XCTestCase

@end

@implementation EHPlainAlertExampleTests

- (void)setUp {
    [super setUp];
    // Put setup code here. This method is called before the invocation of each test method in the class.
}

- (void)tearDown {
    // Put teardown code here. This method is called after the invocation of each test method in the class.
    [super tearDown];
}

- (void)testErrorAlert
{
    NSError * testError = [[NSError alloc] initWithDomain:@"My domain"
                                                 code:1337
                                             userInfo:@{NSLocalizedDescriptionKey: NSLocalizedString(@"Operation was unsuccessful.", nil),
                                                        NSLocalizedFailureReasonErrorKey: NSLocalizedString(@"The operation timed out.", nil),
                                                        NSLocalizedRecoverySuggestionErrorKey: NSLocalizedString(@"Have you tried turning it off and on again?", nil)}];
    
    EHPlainAlert * alert = [EHPlainAlert showError:testError];
    
    XCTAssertNotNil(alert);
    XCTAssertTrue([[alert subtitleString] isEqualToString:testError.localizedDescription]);
    
}

- (void)testDomainErrorAlert
{
    NSError * testError = [[NSError alloc] initWithDomain:@"My domain"
                                                     code:1337
                                                 userInfo:@{NSLocalizedDescriptionKey: NSLocalizedString(@"Operation was unsuccessful.", nil),
                                                            NSLocalizedFailureReasonErrorKey: NSLocalizedString(@"The operation timed out.", nil),
                                                            NSLocalizedRecoverySuggestionErrorKey: NSLocalizedString(@"Have you tried turning it off and on again?", nil)}];
    
    EHPlainAlert * alert = [EHPlainAlert showDomainError:testError];
    
    XCTAssertNotNil(alert);
    XCTAssertTrue([[alert titleString] isEqualToString:testError.domain]);
    XCTAssertTrue([[alert subtitleString] isEqualToString:testError.localizedDescription]);
    

    
}

- (void)testAlertShow
{
    EHPlainAlert * alert = [EHPlainAlert showAlertWithTitle:@"123" message:@"123" type:ViewAlertPanic];
    XCTAssertTrue([alert.view superview] != nil);
}


- (void)testThatItInitAlert
{
    EHPlainAlert * alert = [[EHPlainAlert alloc] initWithTitle:@"My title" message:@"My message" type:ViewAlertError];
    XCTAssertNotNil(alert);
    
    alert = [[EHPlainAlert alloc] initWithTitle:@"My title" message:@"My message" type:ViewAlertUnknown];
    XCTAssertNotNil(alert);
    
    alert = [[EHPlainAlert alloc] initWithTitle:@"My title" message:@"My message" type:ViewAlertSuccess];
    XCTAssertNotNil(alert);
    
    alert = [[EHPlainAlert alloc] initWithTitle:@"My title" message:@"My message" type:ViewAlertInfo];
    XCTAssertNotNil(alert);
   
    alert = [[EHPlainAlert alloc] initWithTitle:@"My title" message:@"My message" type:ViewAlertPanic];
    XCTAssertNotNil(alert);
}


- (void)testBehaviourChange
{
    NSInteger hex = arc4random()%0xffffff;
    UIColor * color = [UIColor colorWithHex:hex];
    [EHPlainAlert updateAlertColor:color forType:ViewAlertError];
    UIFont * font = [UIFont systemFontOfSize:20];
    [EHPlainAlert updateTitleFont:font];
    [EHPlainAlert updateSubTitleFont:font];
    [EHPlainAlert updateNumberOfAlerts:1];
    [EHPlainAlert updateHidingDelay:1];
    [EHPlainAlert updateAlertIcon:nil forType:ViewAlertInfo];
    EHPlainAlert * alert = [[EHPlainAlert alloc] initWithTitle:@"My title" message:@"My message" type:ViewAlertError];
    XCTAssertNotNil(alert);
    [alert show];
    XCTAssertEqual(alert.view.backgroundColor,color);
    XCTAssertEqual(alert.titleFont,font);
    XCTAssertEqual(alert.subTitleFont,font);
    
}

- (void)testAlertPosition
{
    [EHPlainAlert updateAlertPosition:ViewAlertPositionTop];
    EHPlainAlert * alert = [[EHPlainAlert alloc] initWithTitle:@"My title" message:@"My message" type:ViewAlertError];
    XCTAssertNotNil(alert);
    [alert show];
    XCTAssertLessThan(alert.view.frame.origin.y,70);
}

- (void)testColor
{
    NSInteger hex = arc4random()%0xffffff;
    UIColor * color = [UIColor colorWithHex:hex];
    XCTAssertEqual(hex, [color getHexColor]);
    
}


- (void)testPerformanceExample {
    // This is an example of a performance test case.
    [self measureBlock:^{
        
        for (int i = 0; i < 50; i++)
        {
            [EHPlainAlert showAlertWithTitle:[NSString stringWithFormat:@"%i", i] message:@"Alert" type:ViewAlertInfo];
        }
        // Put the code you want to measure the time of here.
    }];
}

@end


EHPlainAlertExample/ViewController.m
//
//  ViewController.m
//  EHPlainAlertExample
//
//  Created by Danila Gusev on 17/03/2016.
//  Copyright © 2016 josshad. All rights reserved.
//

#import "ViewController.h"
#import "EHPlainAlert.h"
@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [EHPlainAlert updateAlertPosition:ViewAlertPositionBottom];
    [EHPlainAlert updateShouldShowCloseIcon:NO];
//    [EHPlainAlert updateTitleFont:[UIFont fontWithName:@"TrebuchetMS" size:18]];
//    [EHPlainAlert updateSubTitleFont:[UIFont fontWithName:@"TrebuchetMS" size:10]];
//    [EHPlainAlert updateNumberOfAlerts:4];
//    [EHPlainAlert updateAlertColor:[UIColor colorWithWhite:0 alpha:0.5] forType:ViewAlertPanic];
//    [EHPlainAlert updateHidingDelay:2.5f];
//    [EHPlainAlert updateAlertIcon:nil forType:ViewAlertInfo]; 
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];

    // Dispose of any resources that can be recreated.
}


- (IBAction)successAlert:(id)sender
{
    [EHPlainAlert showAlertWithTitle:@"Success" message:@"Something works!" type:ViewAlertSuccess];
}

- (IBAction)failureAlert:(id)sender
{
    NSError * error = [[NSError alloc] initWithDomain:@"My domain"
                                                 code:1337
                                             userInfo:@{NSLocalizedDescriptionKey: NSLocalizedString(@"Operation was unsuccessful.", nil),
                                                        NSLocalizedFailureReasonErrorKey: NSLocalizedString(@"The operation timed out.", nil),
                                                        NSLocalizedRecoverySuggestionErrorKey: NSLocalizedString(@"Have you tried turning it off and on again?", nil)}];
    [EHPlainAlert showError:error];
}

- (IBAction)infoAlert:(id)sender
{
    EHPlainAlert * ehAlert = [[EHPlainAlert alloc] initWithTitle:@"Info" message:@"This is info message" type:ViewAlertInfo];
    ehAlert.titleFont = [UIFont fontWithName:@"TrebuchetMS" size:15];
    ehAlert.subTitleFont = [UIFont fontWithName:@"TrebuchetMS-Italic" size:12];
    ehAlert.shouldShowCloseIcon = YES;
    [ehAlert show];

}

- (IBAction)panicAlert:(id)sender
{
    [EHPlainAlert showAlertWithTitle:@"Panic!" message:@"Something brokе!" type:ViewAlertPanic];
}

- (IBAction)infoWithSafari:(id)sender
{
    EHPlainAlert * ehAlert = [[EHPlainAlert alloc] initWithTitle:@"Hmm..." message:@"Tap for information" type:ViewAlertInfo];
    ehAlert.action = ^{
        [[UIApplication sharedApplication] openURL:[NSURL URLWithString:@"https://github.com/josshad/EHPlainAlert"]];
    };
    ehAlert.messageColor = [UIColor blueColor];
    ehAlert.iconImage = nil;
    [ehAlert show];
}

- (IBAction)hideAll:(id)sender
{
    [EHPlainAlert hideAll:YES];
}
@end
